saleor/payment/gateways/adyen/webhooks.py

```diff
@@ -165,6 +166,20 @@ def create_order(payment, checkout, manager):
         discounts = fetch_active_discounts()
         lines = fetch_checkout_lines(checkout)
         checkout_info = fetch_checkout_info(checkout, lines, discounts, manager)
+        checkout_total = calculate_checkout_total_with_gift_cards(
+            manager=manager,
+            checkout_info=checkout_info,
+            lines=lines,
+            address=checkout.shipping_address or checkout.billing_address,
+            discounts=discounts,
+        )
+        # when checkout total value is different than total amount from payments
+        # it means that some products has been removed during the payment was completed
+        if checkout_total.gross.amount != payment.total:
```

Comment:
Maybe I am missing something, but isn't it handled in `complet_checkout`? - `complete_checkout._prepare_checkout.clean_checkout_payment`

Reply:
It is. But this is the situation when something is changed during the payment confirmation like it is in 3D secure. For example, an order is created, payment is in process, someone deletes variant, payment is finished.